crates/webidl/src/idl_type.rs

```diff
@@ -324,14 +327,26 @@ impl<'a> ToIdlType<'a> for Identifier<'a> {
     }
 }
 
+// We default to Float32Array's being mutable, but in certain cases where we're certain that
+// slices won't get mutated on the JS side (such as the WebGL APIs) we might, later in the flow,
+// instead use the immutable version.
+impl<'a> ToIdlType<'a> for term::Float32Array {
+    fn to_idl_type(&self, _record: &FirstPassRecord<'a>) -> IdlType<'a> {
+        IdlType::Float32Array {immutable: false}
+    }
+}
+
 macro_rules! terms_to_idl_type {
     ($($t:tt => $r:tt)*) => ($(
         impl<'a> ToIdlType<'a> for term::$t {
             fn to_idl_type(&self, _record: &FirstPassRecord<'a>) -> IdlType<'a> {
-                IdlType::$r
+                match IdlType::$r {
+                    IdlType::Callback => IdlType::Float32Array { immutable: false },
+                    _ => IdlType::$r
+                }
```

Comment:
Hm this change looks suspicious, could you expand on what's happening here?

Reply:
This is my bad - this is just leftover from when I was trying to get the macro to work with enums that have fields but then I ended up just writing the `impl` right above and didn't need to touch the macro.

Didn't clean this up before I asked for your help... Sorry!